### ISSUE TITLE
Translating Safe to another languages

### ISSUE THREAD

**Issue by pdroalves** [safeapp/safe#81]
It would be very interesting if Safe could be easily translated to another languages.

I gave a look at master branch (I didn't found anything more recent)  and looks like that all phrases are stored in a head file as [constants](https://github.com/safeapp/safe/blob/master/src/safe/constants.h). It is almost trivial to provide different translations at compile time, but impossible to change it as runtime. This way, we would need different binaries to provide different languages. This is a very bad idea.

A much better strategy is to refactor the way Safe loads it's phrases to an dynamic vector or something like it. This way, in runtime becomes possible to switch the entire package of phrases.

**Comment by rianhunter** [comment]
Yeah this sounds good. The original solution was just a stop-gap since i18n wasn't a priority.

In the middle of developing Safe2 so not focused on this at the moment but feel free to send me a PR. Would love to have Safe2 in multiple languages.

**Comment by pdroalves** [comment]
Great. 

I will do that.

**Comment by rianhunter** [comment]
One thing to consider,

The strings may change drastically so focus more on the system than
translating the strings themselves.

I will provide comments on the PR but feel free to ask questions about
implementation direction if you need guidance.

**Comment by pdroalves** [comment]
I'm so sorry for the delay. I couldn't allocate time on my schedule to address this until this week.

I wrote a module to handle xml files with different translations and already moved most of the hard-coded text to a resource file called "en-us.xml". However, I still have three issues to handle before commiting it:

1) the file "src/safe/constants.h" has a lot of hard-coded text assigned to several macros. I can't replace those with runtime calls to the function get_phrase() from my module. How to deal with it?
2) I think I found some hard-coded text in *.xib files for Mac, but I have no idea how to deal with then. 
3) I have been failing completely to setup a Windows environment to compile Safe with my new module. Do you know if it is possible to compile and test it in Linux?

Lastly, should I send the ongoing work on a PR?

**Comment by rianhunter** [comment]
Hi Pedro, I saw your PR, it's awesome. Let's communicate on there, I have some recommendations for the patch.
